src/services/fetchers/PolygonIOStockPriceFetcher.ts

```diff
@@ -1,20 +1,21 @@
 import {inject, injectable} from 'inversify';
 
 import PolygonIOStockPriceService from '../PolygonIOStockPriceService.js';
-import {FeedFetcherInterface} from '../../types/fetchers.js';
+import {FeedFetcherInterface, FeedFetcherOptions} from '../../types/fetchers.js';
 
 @injectable()
 class PolygonIOPriceFetcher implements FeedFetcherInterface {
   @inject(PolygonIOStockPriceService) polygonIOStockPriceService!: PolygonIOStockPriceService;
 
-  async apply({sym}: {sym: string}, timestamp: number): Promise<number> {
-    const price = await this.polygonIOStockPriceService.getLatestPrice(sym, timestamp);
+  async apply(params: {sym: string}, options: FeedFetcherOptions): Promise<number> {
+    // TODO: if it is the latest price, does it need a timestamp as argument?
```

Comment:
to answer this question please analyze how timestamp is used in thie method, if it is passed as argument then most probably it is needed here.

Reply:
In the very end the `getLatestPrice` is calling this method:

```typescript
  /**
   * Gets a value before the provided timestamp
   */
  async value(symbol: string, beforeTimestamp: number): Promise<number | null> {
  ```
  
  So, yes, it is used but I imagine `getLatestPrice` could infere another way of deriving this `beforeTimestamp` without letting the user to decide. Could we leave the TODO and refactor it in the future?